Redesign tables on mobile

jadebuckwalter opened the pull request:
Hide and show columns in the Assignments and Categories tables based on
the screen size. Fixes #203.

jadebuckwalter committed:
Dynamically resize columns

Hide and show columns in the Assignments and Categories tables based on
the screen size.

psvenk committed:
Add category to assignment info modal

The category of an assignment is hidden in the table on small screen
sizes; include this information in the assignment info modal.

psvenk committed:
Make adjustColumns trigger after data loaded

The load event listener creates a race condition which results in the
assignments table header not loading properly on mobile. Instead, call
adjustColumns just after the assignments and categories tables are
populated with data.

Additionally, this commit has adjustColumns take in as a parameter the
table on which it operates, enabling the resize event listener to be
initialized only once the tables are built. This prevents edge case race
conditions and makes the code more generalizable (if we want to use
adjustColumns for, e.g., the schedule table in the future).

tektaxi committed:
start to migrate stats modal to panel

psvenk committed:
Move stats modal under categories table

Instead of occupying the entire screen, have the stats modal sit between
the categories table and the assignments table.

psvenk commented:
I just now made a commit that moves the stats modal to be between the categories and assignments tables, but the styling could use some work. We should also look into giving the other modals the same treatment.

jadebuckwalter committed:
Style stats modal

Style the (no-longer) modal so that it looks more like the surrounding
tables, remove extra padding, and remove animations. We still need to
have the modal close when a different class is selected.

jadebuckwalter committed:
Hide modal when necessary

Close the stats modal when a different class is selected or the data is
reset.

jadebuckwalter committed:
Apply changes to corrections modal

Treat the corrections modal as a stats modal, and have it appear above
the stats modal and behave in the same way.

psvenk reviewed (changes requested):
The assignments table header still does not load properly on mobile for me. Apart from that, this looks good to me.

psvenk reviewed (approved):
I can no longer reproduce the issue with the assignments table header; maybe there was some caching issue going on with my previous test. So, I think we can consider that issue fixed and this pull request ready to merge.
